assets/index.js

```diff
@@ -41,7 +47,53 @@ function updateTimer() {
 		}
 		return response.json()
 	}).then(capturing => {
-		console.debug('capturing', capturing)
+		console.log('capturing', capturing);
+		if(is_active != capturing && !capturing){
+			updateCalendar()
+		}
+		is_active = capturing;
 		updateView(capturing ? config.capturing : config.idle);
 	})
 }
+
+function parseCalendar(active){
+	// Do we want 'Startet/Endet in' or 'Startet/Endet um'?
+	let diff = 0;
+	let t = 0;
+
+	now = Date.now();
+	if (calendar.length > 0){
+		t = is_active ? calendar[0].End : calendar[0].Start;
+		diff = t - now;
+		console.debug('Diff ', diff, t, now);
+	} else {
+		console.debug('Calendar is empty');
+		
+	}
+
+	hours = (diff > 0) ? Math.floor(diff / (1000 * 60 * 60)) : 0;
+	minutes = (diff > 0) ? Math.floor((diff / (1000 * 60)) % 60) : 0;
+	seconds = (diff > 0) ? Math.floor((diff / 1000) % 60) : 0;
+
+	hours = (hours < 10) ? '0' + hours : hours;
+	minutes = (minutes < 10) ? '0' + minutes : minutes;
+	seconds = (seconds < 10) ? '0' + seconds : seconds;
+
+	console.debug('Remaining ', diff/1000);
+	if (calendar.length == 0 && !is_active) {
+		return 'Keine Aufzeichnung geplant';
+	} else {
+		return (hours > 0) ? active.info + ' ' + hours + ':' + minutes + ':' + seconds : active.info + ' ' + minutes + ':' + seconds;
```

Comment:
I would always display hours. Looking at the following image, you don't quite know if it's 17 hours? Sure, you realize that quickly, after you see the time decrease every second, but I still think always having the hours is more common and easier to understand.

![Screenshot from 2024-07-15 22-42-49](https://github.com/user-attachments/assets/7c6ce263-4f68-44ff-aeac-0ad1bde07aef)